### PR TITLE
fix: calendar re-render issue on eventModal

### PR DESCRIPTION
**Pull request by iRohitGaur**
### What is the change?

- fix: calendar used to re-render on eventModal route when entering details
- refactor: removed unnecessary prop drilling from `$eventId.tsx` to `eventModal/index.tsx`

Provide a small description of what did you change and provide the reference to the issue ticket.

### Is it bug?

- Steps to repro
- Expected
- Actual

### \*Dev Tested?

- [ ] Yes
- [ ] No

#### Platforms

- [ ] Web

#### Browsers

- [ ] Chrome
- [ ] Safari
- [ ] Firefox

### Before / After Change Screenshots

> For visual or interaction changes. Can be video / screenshot.

### CODE REVIEW

File: app/components/common/eventModal/index.tsx [hunk 1]
```diff
@@ -22,26 +22,28 @@ import { patchEvent, postEvent } from '../../../constants/urls.constants';
...
+  const [statuses, setStatuses] = useState<{
     creatingPost: 'idle' | 'loading';
   }>({
     creatingPost: 'idle',
   });
+  const params = useParams();
+  const navigate = useNavigate();
   const routerLocation = useLocation() as { state: { start: string; end: string } };
+
+  const currentEvent = calendarEvent.event!;
```
Comment (isVivek99) [inline app/components/common/eventModal/index.tsx:46]: can this break?

Comment (iRohitGaur) [reply app/components/common/eventModal/index.tsx:46]: > can this break?

@isVivek99 EventModal needs the `event` prop and we set it in calendarEvent initially. So no, unless you manually remove the `event` property from the state, it won't break.

File: app/components/common/eventModal/index.tsx [hunk 2]
```diff
@@ -55,21 +57,24 @@ export default function EventModal({ events, currentEvent, setCalendarEvent }: E
...
-  React.useEffect(() => {
+  useEffect(() => {
     if (params.eventId === 'new' && routerLocation.state) {
       setCalendarEvent((e) => ({
         ...e,
-        start: dayjs(routerLocation.state.start),
-        end: dayjs(routerLocation.state.end),
+        event: {
+          ...e.event,
+          start: dayjs(routerLocation.state.start).toDate(),
```
Comment (isVivek99) [inline app/components/common/eventModal/index.tsx:70]: how was this working previously, if we were not converting to date

Comment (iRohitGaur) [reply app/components/common/eventModal/index.tsx:70]: @isVivek99 first of all it was not set inside the `event` property as it should.
After I set it to the event, ts threw the actual error regarding datatype mismatch. That's why the conversion.

File: app/components/common/eventModal/index.tsx (hunk 1 above)
Comment (isVivek99) [reply app/components/common/eventModal/index.tsx:46]: can we remove that `!` mark ?

Comment (iRohitGaur) [reply app/components/common/eventModal/index.tsx:46]: @isVivek99 no, its set as an optional in the interface, so its required.

Comment (isVivek99) [reply app/components/common/eventModal/index.tsx:46]: on line `25`, i can see it as not being optional

**Review by isVivek99 — APPROVED** [review]

**Review by harshith-venkatesh — APPROVED** [review]

**Pull request merged**